app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php

```diff
@@ -98,20 +104,20 @@ protected function validateGuestSubscription()
      * Validates the format of the email address
      *
      * @param string $email
-     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws LocalizedException
+     * @throws \Zend_Validate_Exception
      * @return void
      */
     protected function validateEmailFormat($email)
     {
-        if (!\Zend_Validate::is($email, \Magento\Framework\Validator\EmailAddress::class)) {
-            throw new \Magento\Framework\Exception\LocalizedException(__('Please enter a valid email address.'));
+        if (!(new \Zend_Validate)->is($email, EmailAddress::class)) {
```

Comment:
In all the places Magento uses Zend_Validate::is(), but in this case you suggest to replace it with object creation.
https://github.com/magento/magento2/search?p=3&q=Zend_Validate&unscoped_q=Zend_Validate
In general I agree that using static methods not the best practice, but it's better to change it in all places, also not sure if this variant is really optimal.
If you would like to propose replacement to object creation - feel free to create separate PR, but in scope of this PR - please revert this change

Reply:
Thank you for the feedback, I only changed it because Codacy mentioned it as a new issue from my commit (because of the import statements).

Reply:
We're using codaxy just as additional information, not all issues we want to be fixed. As example this one :)